src/UI/Root/Components/PageFrame/PageInitializer.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from "react";
+import { RemoteData } from "@/Core";
+import { DependencyContext } from "@/UI/Dependency";
+
+interface Props {
+  environment: string;
+}
+
+export const PageInitializer: React.FC<Props> = ({ children, environment }) => {
```

Comment:
Why is children needed? Is this to keep things generic?

Reply:
I feel like this does everything needed:

```
export const PageInitializer: React.FC<Props> = ({ environment }) => {
  const { queryResolver } = useContext(DependencyContext);

  queryResolver.useOneTime<"GetEnvironmentSettings">({
    kind: "GetEnvironmentSettings",
    environment: environment,
  });
  return null;
};

```

Reply:
It was because at first I intended to use it as a wrapper, but then I realized that it wasn't needed, so you're right, it can be even simpler. :+1: